gc/base/TLHAllocationSupport.cpp

```diff
@@ -419,4 +433,48 @@ MM_TLHAllocationSupport::objectAllocationNotify(MM_EnvironmentBase *env, void *h
 }
 #endif /* OMR_GC_OBJECT_ALLOCATION_NOTIFY */
 
+/**
+ * Reserve part (top) of TLH for GC if collector requires
+ */
+void
+MM_TLHAllocationSupport::reserveTLHTopForGC(MM_EnvironmentBase *env)
+{
+	uintptr_t bytesToReserveForGC = env->getExtensions()->getGlobalCollector()->reserveAllocCacheSize();
```

Comment:
It can't be const as its dependent on SATB state: `virtual uintptr_t reservedForGCAllocCacheSize() { return (_extensions->isSATBBarrierActive() ? OMR_MINIMUM_OBJECT_SIZE : 0); }`